File: src/student-ui/to react/lib/screens/home/statistics_section.dart
import 'package:englishfluencyguide/models/achievement_model.dart';
import 'package:flutter/material.dart';
import 'package:englishfluencyguide/models/user_model.dart';
import 'package:englishfluencyguide/providers/achievement_provider.dart';
import 'package:englishfluencyguide/l10n/app_localizations.dart';

class StatisticsSection extends StatefulWidget {
  final UserModel? user;
  final AchievementProvider achievementProvider;

  const StatisticsSection({
    Key? key,
    required this.user,
    required this.achievementProvider,
  }) : super(key: key);

  @override
  State<StatisticsSection> createState() => _StatisticsSectionState();
}

class _StatisticsSectionState extends State<StatisticsSection> {
  late List<UserAchievementModel> _achievements = [];
  @override
  void initState() {
    super.initState();
    _loadAchievements();
  }

  Future<void> _loadAchievements() async {
    await widget.achievementProvider.loadUserAchievements(widget.user!.uid);
    _achievements = widget.achievementProvider.userAchievements;
    print("_achievements $_achievements");
    if (mounted) setState(() {});
  }

  @override
  Widget build(BuildContext context) {
    // _loadAchievements();

    int seconds = widget.user?.totalStudyMinutes ?? 0;
    int minutes = (seconds / 60).round();
    final theme = Theme.of(context);
    final l10n = AppLocalizations.of(context)!;

    return Row(
      children: [
        Expanded(
          child: _EnhancedStatCard(
            icon: Icons.timer_outlined,
            value: minutes.toString(),
            label: l10n.minutesToday,
            color: Colors.blue,
          ),
        ),
        const SizedBox(width: 12),
        Expanded(
          child: _EnhancedStatCard(
            icon: Icons.emoji_events_outlined,
            value: '${widget.achievementProvider.userAchievements.length}',
            label: l10n.achievements,
            color: Colors.amber,
          ),
        ),
        const SizedBox(width: 12),
        Expanded(
          child: _EnhancedStatCard(
            icon: Icons.star_outline,
            value: '${5}',
            label: l10n.xpPoints,
            color: Colors.purple,
          ),
        ),
      ],
    );
  }
}

class _EnhancedStatCard extends StatelessWidget {
  final IconData icon;
  final String value;
  final String label;
  final Color color;

  const _EnhancedStatCard({
    required this.icon,
    required this.value,
    required this.label,
    required this.color,
  });

  @override
  Widget build(BuildContext context) {
    final theme = Theme.of(context);

    return Container(
      padding: const EdgeInsets.all(20),
      decoration: BoxDecoration(
        color: theme.colorScheme.surface,
        borderRadius: BorderRadius.circular(16),
        border: Border.all(
          color: theme.colorScheme.outline.withOpacity(0.1),
        ),
        boxShadow: [
          BoxShadow(
            color: theme.colorScheme.shadow.withOpacity(0.05),
            blurRadius: 8,
            offset: const Offset(0, 2),
          ),
        ],
      ),
      child: Column(
        children: [
          Container(
            padding: const EdgeInsets.all(12),
            decoration: BoxDecoration(
              color: color.withOpacity(0.1),
              borderRadius: BorderRadius.circular(12),
            ),
            child: Icon(icon, size: 24, color: color),
          ),
          const SizedBox(height: 12),
          Text(
            value,
            style: theme.textTheme.headlineSmall?.copyWith(
              fontWeight: FontWeight.bold,
              color: theme.colorScheme.onSurface,
            ),
          ),
          const SizedBox(height: 4),
          Text(
            label,
            style: theme.textTheme.bodySmall?.copyWith(
              color: theme.colorScheme.onSurfaceVariant,
            ),
            textAlign: TextAlign.center,
          ),
        ],
      ),
    );
  }
}

File: src/student-ui/students-pages/student-dashboard-page/components/StatisticsSection.js
import React from "react";
import { Grid, Card, Box, Typography } from "@mui/material";
import TimerOutlinedIcon from "@mui/icons-material/TimerOutlined";
import EmojiEventsOutlinedIcon from "@mui/icons-material/EmojiEventsOutlined";
import StarOutlineIcon from "@mui/icons-material/StarOutline";

const StatCard = ({ icon, value, label, color }) => (
  <Card
    sx={{
      p: 2.5,
      borderRadius: 2,
      display: "flex",
      flexDirection: "column",
      alignItems: "center",
      boxShadow: 1,
      bgcolor: "background.paper",
    }}
  >
    <Box
      sx={{
        p: 1.5,
        bgcolor: color + "22",
        borderRadius: 2,
        mb: 1.5,
        display: "flex",
        alignItems: "center",
        justifyContent: "center",
      }}
    >
      {icon}
    </Box>
    <Typography variant="h6" fontWeight={700} color="text.primary">
      {value}
    </Typography>
    <Typography variant="body2" color="text.secondary" align="center">
      {label}
    </Typography>
  </Card>
);

const StatisticsSection = ({
  minutesToday = 0,
  achievementsCount = 0,
  xpPoints = 0,
}) => (
  <Grid container spacing={2} sx={{ mb: 3 }}>
    <Grid item xs={12} sm={4}>
      <StatCard
        icon={<TimerOutlinedIcon sx={{ color: "#1976d2" }} />}
        value={minutesToday}
        label="Minutes Today"
        color="#1976d2"
      />
    </Grid>
    <Grid item xs={12} sm={4}>
      <StatCard
        icon={<EmojiEventsOutlinedIcon sx={{ color: "#ffb300" }} />}
        value={achievementsCount}
        label="Achievements"
        color="#ffb300"
      />
    </Grid>
    <Grid item xs={12} sm={4}>
      <StatCard
        icon={<StarOutlineIcon sx={{ color: "#8e24aa" }} />}
        value={xpPoints}
        label="XP Points"
        color="#8e24aa"
      />
    </Grid>
  </Grid>
);

export default StatisticsSection;
